jsettlers.logic/src/main/java/jsettlers/ai/highlevel/AiStatistics.java

```diff
@@ -277,13 +381,27 @@ private void updateFreeLand(short x, short y) {
 				sortedCuttableObjectsInDefaultPartition.put(STONE, stones);
 			}
 			stones.addNoCollission(x, y);
+			updateNearStones(x, y);
 		}
 		ELandscapeType landscape = landscapeGrid.getLandscapeTypeAt(x, y);
 		if (landscape.isRiver()) {
 			sortedRiversInDefaultPartition.addNoCollission(x, y);
 		}
 	}
 
+	private void updateNearStones(short x, short y) {
+		for (EDirection dir : EDirection.VALUES) {
+			int dx = dir.getNextTileX(x, NEAR_STONE_DISTANCE);
+			int dy = dir.getNextTileY(y, NEAR_STONE_DISTANCE);
+			if (mainGrid.isInBounds(dx, dy)) {
+				byte playerId = partitionsGrid.getPlayerIdAt(dx, dy);
+				if (playerId != -1  && hasPlayersBlockedPartition(playerId, x, y)) {
+					playerStatistics[playerId].stonesNearBy.addNoCollission(x, y);
```

Comment:
Am I correct that you add any stones in that area multiple times? I mean, I stone has a ground area larger 1 and therefore will be added multiple times. Might this cause problems? Or is it just not optimal?